### DISCUSSION TITLE
File in Storage not being overwritten

### DISCUSSION THREAD

**Discussion by zacherygentry** [supabase/supabase#2289] [Questions] [answered]
I have code that looks like this
```
const path = `${user.id}/profile-pic.png`;

const { data, error } = await supabase.storage
  .from('profiles')
  .upload(path, profile_picture, {
    upsert: true,
  });
```

This works great the first time it is run. However, if I try running it again with a different image, the file in storage does not change. Any ideas?

**Comment by inian** [comment, marked as answer]
Hey @zacherygentry, hmm that shouldn't be happening. Do you get any error when you do the upsert the second time?

**Reply by zacherygentry** [reply]
I got it working. I believe it was working all along but when viewing the image from the supabase dashboard, it would show the same image no matter what. After actually downloading the image, it was the correct one. Some sort of caching issue?

**Reply by inian** [reply]
Ah yeah by default assets uploaded have a [cache time](https://github.com/supabase/storage-js/blob/d7721b85008ffca286a14159db1038a8812c86c1/src/lib/StorageFileApi.ts#L15) of 3600 seconds unless it is overriden when uploading
